cmake/emu/nsim.cmake

```diff
@@ -1,63 +1,95 @@
 # SPDX-License-Identifier: Apache-2.0
-if(${CONFIG_SOC_NSIM_HS_SMP})
-# mdb is required to run nsim multicore targets
-find_program(
-  MDB
-  mdb
+if("${BOARD_DEBUG_RUNNER}" STREQUAL "mdb-nsim" OR "${BOARD_FLASH_RUNNER}" STREQUAL "mdb-nsim")
+  # mdb is required to run nsim multicore targets
+  find_program(
+    MDB
+    mdb
   )
+  set(MDB_BASIC_OPTIONS -nooptions -nogoifmain -toggle=include_local_symbols=1)
 
-if(${CONFIG_SOC_NSIM_HS_SMP})
- set(MDB_ARGS mdb_hs_smp.args)
-endif()
+  if(${CONFIG_SOC_NSIM_HS_SMP})
+    set(MDB_ARGS mdb_hs_smp.args)
+  elseif(${CONFIG_SOC_NSIM_EM})
+    set(MDB_ARGS mdb_em.args)
+  elseif(${CONFIG_SOC_NSIM_EM7D_V22})
+    set(MDB_ARGS mdb_em7d_v22.args)
+  elseif(${CONFIG_SOC_NSIM_SEM})
+    set(MDB_ARGS mdb_sem.args)
+  elseif(${CONFIG_SOC_NSIM_HS})
+    set(MDB_ARGS mdb_hs.args)
+  elseif(${CONFIG_SOC_NSIM_HS6X})
+    set(MDB_ARGS mdb_hs6x.args)
+  endif()
+
+  # remove previous .sc.project folder which has temporary settings for MDB.
+  set(MDB_OPTIONS rm -rf ${APPLICATION_BINARY_DIR}/.sc.project &&)
+  if(CONFIG_MP_NUM_CPUS GREATER 1)
+    set(MULTIFILES ${MDB} -multifiles=core0)
+    foreach(val RANGE ${CONFIG_MP_NUM_CPUS})
+      if(val LESS CONFIG_MP_NUM_CPUS)
+        MATH(EXPR PSET_NUM "${val}+1")
+        if (val GREATER 0)
+          list(APPEND MDB_OPTIONS &&)
+        endif()
+        list(APPEND MDB_OPTIONS ${MDB} -pset=${PSET_NUM} -psetname=core${val})
+        if (val GREATER 0)
+          list(APPEND MDB_OPTIONS -prop=download=2)
+          set(MULTIFILES ${MULTIFILES},core${val})
```

Comment:
See https://github.com/zephyrproject-rtos/zephyr/pull/36535 on how we order cores (`core0` to be added in the end - rationale is in @IRISZZW's PR).